src/hypothesis/internal/charmap.py

```diff
@@ -106,6 +107,36 @@ def categories():
     return tuple(_categories)
 
 
+def validate_categories(cats, name=None):
+    """Return a tuple of Unicode categories in a normalised order.
+
+    This function expands one-letter designations of a major class to include
+    all subclasses:
+
+    >>> validate_categories(['N'])
+    ('Nd', 'Nl', 'No')
+
+    See section 4.5 of the Unicode standard for more on classes:
+    https://www.unicode.org/versions/Unicode10.0.0/ch04.pdf
+
+    If the collection ``cats`` includes any elements that do not represent a
+    major class or a class with subclass, InvalidArgument is raised.
+
+    """
+    if cats is None:
+        return None
+    major_classes = ('L', 'M', 'N', 'P', 'S', 'Z', 'C')
```

Comment:
Did major classes like this work before? I don't think they did. If not, this should be mentioned in the changelog and should have some accompanying tests.